Gemm impl compilation errors (OpenMP backend) in some nightlies - Power8+gcc7.2 w/cpp17, Clang10

Following merge of #935 the following nightly builds began failing:

1. White testbed - OpenMP_gcc_720_cpp17

Error messages:
```
23:36:25 In file included from /home/jenkins/white/workspace/KokkosKernels_White_OpenMP_gcc_720_cpp17/kokkos-kernels/src/blas/impl/KokkosBlas3_gemm_spec.hpp:52:0,
23:36:25                  from /home/jenkins/white/workspace/KokkosKernels_White_OpenMP_gcc_720_cpp17/TestAll_2021-06-01_23.31.54/gcc/7.2.0/OpenMP-release/src/impl/generated_specializations_cpp/gemm/Blas3_gemm_eti_DOUBLE_LAYOUTLEFT_EXECSPACE_OPENMP_MEMSPACE_HOSTSPACE.cpp:48:
23:36:25 /home/jenkins/white/workspace/KokkosKernels_White_OpenMP_gcc_720_cpp17/kokkos-kernels/src/blas/impl/KokkosBlas3_gemm_impl.hpp: In lambda function:
23:36:25 /home/jenkins/white/workspace/KokkosKernels_White_OpenMP_gcc_720_cpp17/kokkos-kernels/src/blas/impl/KokkosBlas3_gemm_impl.hpp:438:3: sorry, unimplemented: unexpected AST of kind omp_simd
23:36:25    });
23:36:25    ^
23:36:25 /home/jenkins/white/workspace/KokkosKernels_White_OpenMP_gcc_720_cpp17/kokkos-kernels/src/blas/impl/KokkosBlas3_gemm_impl.hpp:438: confused by earlier errors, bailing out
23:36:25 make[2]: *** [src/CMakeFiles/kokkoskernels.dir/impl/generated_specializations_cpp/gemm/Blas3_gemm_eti_DOUBLE_LAYOUTLEFT_EXECSPACE_OPENMP_MEMSPACE_HOSTSPACE.cpp.o] Error 1
```

Reproducer instructions:
```
module load gcc/7.2.0 cmake/3.19.3

$KOKKOSKERNELS_PATH/cm_generate_makefile.bash --with-devices=OpenMP --arch=Power8 --compiler=g++ --cxxflags="-O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized " --cxxstandard="17" --kokkos-path=$KOKKOS_PATH --kokkoskernels-path=$KOKKOSKERNELS_PATH --with-scalars='double,complex_double' --with-ordinals=int --with-offsets=int,size_t --with-layouts=LayoutLeft --no-examples 
```

2. kokkos-dev-2 testbed - clang/10 with OpenMP

Error messages:
```
06:05:05 In file included from /home/jenkins/slave/workspace/KokkosKernels_KokkosDev2_CLANG10/kokkos-kernels/src/blas/impl/KokkosBlas3_gemm_spec.hpp:52:
06:05:05 /home/jenkins/slave/workspace/KokkosKernels_KokkosDev2_CLANG10/kokkos-kernels/src/blas/impl/KokkosBlas3_gemm_impl.hpp:411:63: error: loop not vectorized: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Werror,-Wpass-failed=transform-warning]
06:05:05   Kokkos::parallel_for(Kokkos::TeamThreadRange(team,blockA0), [&] (const int i) {
```

Reproducer
```
module load sems-clang/10.0.0 sems-cmake/3.17.1
module swap sems-gcc/5.3.0 sems-gcc/8.3.0

$KOKKOSKERNELS_PATH/cm_generate_makefile.bash --with-devices=OpenMP --arch=SNB,Volta70 --compiler=clang++ --cxxflags="-O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized " --cxxstandard="14" --kokkos-path=$KOKKOS_PATH --kokkoskernels-path=$KOKKOSKERNELS_PATH --with-scalars='double,complex_double' --with-ordinals=int --with-offsets=int,size_t --with-layouts=LayoutLeft 
```

@e10harvey do you have time to take a look?

--- a/src/KokkosKernels_Macros.hpp
+++ b/src/KokkosKernels_Macros.hpp
@@ -62,11 +62,14 @@
 
 #if !defined(KOKKOS_ENABLE_CUDA) && !defined(KOKKOS_ENABLE_HIP) && \
     defined(KOKKOS_ENABLE_OPENMP)
-#if defined(KOKKOS_COMPILER_GNU)
+// For clang OpenMP support, see
+// https://clang.llvm.org/docs/OpenMPSupport.html#id1
+#if defined(KOKKOS_COMPILER_GNU) || defined(KOKKOS_COMPILER_CLANG)
 // GCC 4.8.5 and older do not support #pragma omp simd
 #if (KOKKOS_COMPILER_GNU > 485)
 #define KOKKOSKERNELS_ENABLE_OMP_SIMD
 #endif
+// TODO: Check for a clang version that supports #pragma omp simd
 #else
 // All other Kokkos-supported compilers support it.
 #define KOKKOSKERNELS_ENABLE_OMP_SIMD

--- a/src/KokkosKernels_Macros.hpp
+++ b/src/KokkosKernels_Macros.hpp
@@ -66,7 +66,8 @@
 // https://clang.llvm.org/docs/OpenMPSupport.html#id1
 #if defined(KOKKOS_COMPILER_GNU) || defined(KOKKOS_COMPILER_CLANG)
 // GCC 4.8.5 and older do not support #pragma omp simd
-#if (KOKKOS_COMPILER_GNU > 485)
+// Do not enable when using GCC 7.2.0 + C++17 due to a bug in gcc
+#if (KOKKOS_COMPILER_GNU > 485) && !(KOKKOS_COMPILER_GNU == 720 && defined(KOKKOS_ENABLE_CXX17))
 #define KOKKOSKERNELS_ENABLE_OMP_SIMD
 #endif
 // TODO: Check for a clang version that supports #pragma omp simd

